### ISSUE TITLE
Optional slice

### ISSUE THREAD

**Issue by grafael** [samsarahq/thunder#203]
It's not possible to create a schema with optional **array** (slice). So far I've been using pointers to let my struct fields optional and works preety well. I had to create a new struct to encapsulate my slice in order to make it optional. There are any way to make it directly?

**Comment by mikehawkes** [comment]
Thanks for this comment - it brought a few hours of misery to an end! I think this works well but just needed documentation - mandatory is an object, optional a pointer to an object ...

**Comment by JakeWilliams22** [comment]
Agreed - seems like there should be a way to create optional fields without making them pointers
